### PR TITLE
Tests: Improve tests

### PR DESCRIPTION
**Pull request by Kojoley**
- Split Spirit v2 tests
  - Readd missing v2 tests
  - Run X3 tests
  - Remove `-p3` suffix
  - Use `<c++-template-depth>`
  - Automatic adding of subproject name prefix
  - Flatten tests declaration

### CODE REVIEW

**Comment by Kojoley** [comment]
It is now should be safe to merge as https://github.com/boostorg/build/commit/6cea15c958ba3a5a60d6de40236a1f2cdbde20ae landed.
At least `b2 status/ -n -a -d0 --dump-tests --reconfigure` now correctly shows all the Spirit tests.

File: test/x3/Jamfile [hunk 1]
```diff
@@ -8,107 +8,99 @@
...
+run lit1.cpp ;
+run lit2.cpp ;
+run list.cpp ;
+run matches.cpp ;
+run no_case.cpp ;
+run no_skip.cpp ;
+run not_predicate.cpp ;
+run optional.cpp ;
+run plus.cpp ;
+run with.cpp ;
```
Comment (octopus-prime) [inline test/x3/Jamfile]: Where is container_support.cpp ??

File: test/x3/Jamfile [hunk 2]
```diff
@@ -8,107 +8,99 @@
...
+        <c++-template-depth>512
         <toolset>gcc:<cxxflags>-std=c++1y
-        <toolset>gcc:<cxxflags>-ftemplate-depth-512
         <toolset>clang:<cxxflags>-std=c++1y
-        <toolset>clang:<cxxflags>-ftemplate-depth-512
         <toolset>darwin:<cxxflags>-std=c++1y
-        <toolset>darwin:<cxxflags>-ftemplate-depth-512
-    :
-    :
     ;
```
Comment (octopus-prime) [inline test/x3/Jamfile]: Why there is `-std=c++1y`? The c++ dialect is inject from outside into the test, isn't it?

File: test/qi/Jamfile [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+run rule3.cpp ;
+run rule4.cpp ;
+run sequence.cpp ;
+run sequential_or.cpp ;
+run skip.cpp ;
+run stream.cpp ;
+run symbols1.cpp ;
+run symbols2.cpp ;
+run terminal_ex.cpp ;
+run tst.cpp /boost//system ;
```
Comment (octopus-prime) [inline test/qi/Jamfile]: Why there is still useless dependency to boost-system?

Comment (Kojoley) [reply test/qi/Jamfile]: I know at least one toolset it requires - MinGW.

File: test/x3/Jamfile (hunk 1 above)
Comment (Kojoley) [reply test/x3/Jamfile]: There are some other missed tests. I did not catch up all of them, it is a part of the other PR.

File: test/x3/Jamfile (hunk 2 above)
Comment (Kojoley) [reply test/x3/Jamfile]: It is requirements, but actually they seems not to work as expected. I did not wrote it, but fixing this was not a plan for this PR.

File: test/x3/Jamfile (hunk 1 above)
Comment (octopus-prime) [reply test/x3/Jamfile]: Which other PR?
Is there a ticket for adding kicked tests again - containing a list of kicked test to add later again?

Big chance to forget/miss some of the kicked tests...

**Pull request merged**